plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java

```diff
@@ -358,7 +364,11 @@ public LocationEngine getLocationEngine() {
   @Nullable
   @RequiresPermission(anyOf = {ACCESS_FINE_LOCATION, ACCESS_COARSE_LOCATION})
   public Location getLastKnownLocation() {
-    return locationEngine != null ? locationEngine.getLastLocation() : null;
+    Location location = locationEngine != null ? locationEngine.getLastLocation() : null;
+    if (location == null) {
+      location = lastLocation;
```

Comment:
@LukasPaczos Per javadoc above - here we are trying to get the last known location from the location engine and if nothing is found, using `lastLocation` which could also possibly be null?  Just clarifying here, I think it makes sense to add the extra check in case a developer was forcing `Location` updates.

Reply:
Yes, `lastLocation` can also be null if no updates have been pushed to the plugin yet. This is our last resort if `locationEngine` is `null` or fails to provide `non-null` location, like a scenario when we are only forcing location updates.

The idea here is, that `LocationEngine` can be in a position to provide last location (or just more recent location) even if no location updates have been pushed to the plugin itself.